src/context.cpp

```diff
@@ -382,42 +382,99 @@ namespace librealsense
     class dds_sensor_proxy : public software_sensor
     {
     public:
-        dds_sensor_proxy( std::string name, software_device* owner,
-                          std::shared_ptr< realdds::dds_device > const & dev) :
-            software_sensor(name, owner),
-            _dev(dev),
-            _name(name)
+        dds_sensor_proxy( std::string const & sensor_name,
+                          software_device * owner,
+                          std::shared_ptr< realdds::dds_device > const & dev )
+            : software_sensor( sensor_name, owner )
+            , _dev( dev )
+            , _name( sensor_name )
+        {
+        }
+
+        int add_dds_stream( std::shared_ptr< realdds::dds_stream > const & stream )
+        {
+            int stream_index = static_cast< int >( _streams.size() );
+            _streams.push_back( stream );
+            return stream_index;
+        }
+
+        std::shared_ptr< realdds::dds_video_stream_profile >
+        find_profile( int stream_index, realdds::dds_video_stream_profile const & profile )
         {
+            if( stream_index < 0 || stream_index >= _streams.size() )
+            {
+                LOG_ERROR( "Invalid stream index " << stream_index << " in rs2 profile for sensor " << get_name() );
+            }
+            else
+            {
+                for( auto & sp : _streams[stream_index]->profiles() )
+                {
+                    auto vsp = std::static_pointer_cast< realdds::dds_video_stream_profile >( sp );
```

Comment:
Need to check if stream type is video

Reply:
Right now it assumes the streams are all uniform: they're created from a `dds_device` which gets its streams from a `stream-header` that defines the type of its profiles.